### PR TITLE
Integrate compatibility for strict mode

### LINKED ISSUE #5: Make the code run in perl Strict Mode
Cleanup, declaration of variables, etc.

### PR DESCRIPTION
**Pull request by nonnymoose**
It should be ready!
@lezsakdomi, could you look over the code and test it too? I want to make sure I didn't miss anything when testing.

Once merged, this will finally resolve #5.

### CODE REVIEW

File: xsr [hunk 1]
```diff
@@ -65,149 +68,160 @@ elsif (@ARGV == 0) {
...
 # From here on out the code is inspired by this answer: https://unix.stackexchange.com/a/129171, posted by Stéphane Chazelas
-open "X", "-|", "xmodmap -pke" or die "Couldn't open keymap."; # open keymap
-while (<X>) {
+open(my $X, "-|", "xmodmap -pke") or die("Couldn't open keymap."); # open keymap
+my %k;
+my %K;
+while (<$X>) {
 	if (/^keycode\s+(\d+) = (\w+)(?:\s+(\w+))?/) { # keycode <number> = <normal> <shift> ...
 		$k{$1}=$2; # store normal in k, a hash
 		$K{$1}=$3 if $3; # store shift in K, a hash
```
Comment (lezsakdomi) [inline xsr]: I think, these comments are more needed in line 200-201. (new code)

**Comment by lezsakdomi** [comment]
I've tested it, and found no problem. But as I read the code, I don't think so, that could be anything broken.

File: xsr [hunk 2]
```diff
@@ -65,149 +68,160 @@ elsif (@ARGV == 0) {
...
+my $outfilename_noext = $outfilename;
 $outfilename_noext =~ s/\.[^\.]*$//;
 
-if (%mimetypes{$imgext}) {$mimetype = %mimetypes{$imgext};}
+my $mimetype;
+if ($mimetypes{$imgext}) {$mimetype = $mimetypes{$imgext};}
 else {
-	$accepted = "";
-	die "Invalid file type $imgext.\nAccepted file types are:\n" . join(", ", keys %mimetypes) . "\n";
+	# $accepted = ""; what was that for again? :)
```
[the remark this answers is not in the document]
Comment (nonnymoose) [reply xsr]: I have no idea, so I think I'll just remove it.

File: xsr [hunk 3]
```diff
@@ -65,149 +68,160 @@ elsif (@ARGV == 0) {
...
 
 sub convertToB64 {
 	if (! -e $_[0]) {
-		warn "Cannot convert $_[0]: no such file or directory\n";
+		warn("Cannot convert $_[0]: no such file or directory\n");
 		return $_[0];
 	}
 	# return `base64 -w 0 $_[0]`; # -w 0 : do not wrap  # should use internal base64
-	open INIMG, "<", $_[0] or (warn "Cannot open $_[0], this image will be broken!" and return "");
+	open(my $INIMG, "<", $_[0]) or (warn("Cannot open $_[0], this image will be broken!") and return "");
```
[the remark this answers is not in the document]
Comment (nonnymoose) [reply xsr]: I am actually not trying to optimize using and and or; I just used them because it's more convenient than using an entire if statement.

**Pull request merged**